### PR TITLE
Include Readme.md in packages

### LINKED ISSUE #57729: Produce NuGet package READMEs
There are [several package REAMDE](https://github.com/search?q=repo%3Adotnet%2Faspnetcore%20PACKAGE.md&type=code) files in this repository, but they aren't currently being published on NuGet.

We could adopt a similar approach to what [`dotnet/runtime`](https://github.com/dotnet/runtime/blob/6092b21d6e74a343f6019551c64723d7aced7c2e/eng/packaging.targets) does to achieve this.

Prior discussion: https://github.com/dotnet/aspnetcore/pull/54492

### PR DESCRIPTION
**Pull request by wtgodbe**
Fixes https://github.com/dotnet/aspnetcore/issues/57729

test build: https://dev.azure.com/dnceng/internal/_build/results?buildId=2536201&view=results

### CODE REVIEW

**Comment by wtgodbe** [comment]
Test build worked, `Microsoft.AspNetCore.Authentication.JwtBearer` now has a `Package.md`, while other packages without a checked in Readme have nothing

File: Directory.Build.props [hunk 1]
```diff
@@ -239,6 +239,20 @@
     <ArchiveExtension Condition="'$(TargetOsName)' == 'win'">.zip</ArchiveExtension>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <!-- PackageReadmeFile specifies the package readme file name in the package. PackageReadmeFilePath points to the package readme file on disk. -->
+    <EnableDefaultPackageReadmeFile Condition="'$(EnableDefaultPackageReadmeFile)' == '' and '$(IsShipping)' != 'false'">true</EnableDefaultPackageReadmeFile>
+    <PackageReadmeFilePath Condition="'$(PackageReadmeFilePath)' == '' and '$(EnableDefaultPackageReadmeFile)' == 'true'">PACKAGE.md</PackageReadmeFilePath>
+    <PackageReadmeFile Condition="'$(PackageReadmeFile)' == '' and '$(EnableDefaultPackageReadmeFile)' == 'true' and Exists('$(PackageReadmeFilePath)' )">PACKAGE.md</PackageReadmeFile>
+    <!-- Uncomment this once all packages have ReadMes -->
+    <!-- <BeforePack>$(BeforePack);ValidatePackageReadmeExists</BeforePack> -->
```
[the remark this answers is not in the document]
Comment (MackinnonBuck) [reply Directory.Build.props:248]: This one: https://github.com/dotnet/aspnetcore/issues/48392

Comment (MackinnonBuck) [inline Directory.Build.props:248]: It doesn't look like the `ValidatePackageReadmeExists` target is defined in `aspnetcore`. The runtime repo defines it themselves [here](https://github.com/dotnet/runtime/blob/1231a9addd0fb3cfb806c2a89cb45a3018c48bb2/eng/packaging.targets#L336), should we be doing the same? Or does it come from somewhere else?

Comment (MackinnonBuck) [reply Directory.Build.props:248]: Or is the plan to also wait to add that target until we're ready to uncomment the line?

**Review by MackinnonBuck — APPROVED** [review]
Looks great, just one question

File: Directory.Build.props (hunk 1 above)
Comment (wtgodbe) [reply Directory.Build.props:248]: Oh, good catch, I'll just remove it in favor of the defensive conditions I already added

**Pull request merged**